tableaudocumentapi/datasource.py

```diff
@@ -207,6 +208,15 @@ def name(self):
     def version(self):
         return self._version
 
+    @property
+    def caption(self):
+        return self._caption
+
+    @caption.setter
```

Comment:
Are all datasources guaranteed to have a caption or would you ever want to remove it?

We could apply the `None` strategy like we do for port